cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -25,12 +25,15 @@ configure_file(arrow-flight.pc.in "${CMAKE_CURRENT_BINARY_DIR}/arrow-flight.pc"
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/arrow-flight.pc"
         DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
 
+set(ARROW_OPENSSL_LIBS OpenSSL::Crypto OpenSSL::SSL)
+
 set(ARROW_FLIGHT_STATIC_LINK_LIBS
     protobuf::libprotobuf
     gRPC::grpc++
     gRPC::grpc
     gRPC::gpr
-    c-ares::cares)
+    c-ares::cares
+    ${ARROW_OPENSSL_LIBS})
```

Comment:
I'm not tested yet but the following may fix this:

```diff
diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index ea6aa74c..d9074081 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -561,9 +561,9 @@ if(ARROW_WITH_URIPARSER)
 endif()
 
 if(ARROW_USE_OPENSSL)
-  list(APPEND ARROW_LINK_LIBS OpenSSL::Crypto)
-  list(APPEND ARROW_STATIC_LINK_LIBS OpenSSL::Crypto)
-  list(APPEND ARROW_STATIC_INSTALL_INTERFACE_LIBS OpenSSL::Crypto)
+  list(APPEND ARROW_LINK_LIBS OpenSSL::Crypto OpenSSL::SSL)
+  list(APPEND ARROW_STATIC_LINK_LIBS OpenSSL::Crypto OpenSSL::SSL)
+  list(APPEND ARROW_STATIC_INSTALL_INTERFACE_LIBS OpenSSL::Crypto OpenSSL::SSL)
 endif()
 
 if(ARROW_WITH_BROTLI)
```

Reply:
You're right, I've removed the Flight-specific linking